components/miscShared/fn_applyInsignia.sqf

```diff
@@ -5,61 +5,49 @@ params ["_unit"];
 
 LOCAL_ONLY(_unit);
 
-if !EXISTS(f_dict_insigniaColours) then
-{
-    _keyValues =
-    [
-        [str RED,       "insignia_GI_B_RED"],
-        [str ORANGE,    "insignia_GI_B_ORANGE"],
-        [str YELLOW,    "insignia_GI_B_YELLOW"],
-        [str GREEN,     "insignia_GI_B_GREEN"],
-        [str BLUE,      "insignia_GI_B_BLUE"],
-        [str CYAN,      "insignia_GI_B_CYAN"],
-        [str MAGENTA,   "insignia_GI_B_MAGENTA"],
-        [str PINK,      "insignia_GI_B_PINK"],
-        [str PURPLE,    "insignia_GI_B_PURPLE"],
-        [str WHITE,     "insignia_GI_B_WHITE"],
-        [str LIGHTGREY, "insignia_GI_B_LGREY"],
-        [str DARKGREY,  "insignia_GI_B_DGREY"],
-        [str BLACK,     "insignia_GI_B_BLACK"],
-        [str GREY,      "insignia_GI_B_GREY"],
-        [str BROWN,     "insignia_GI_B_BROWN"],
-        [str KHAKI,      "insignia_GI_B_TAN"]
-    ];
-
-    DICT_CREATE_VALS(f_dict_insigniaColours,_keyValues);
-
-};
-
 
 _faction = toLower (faction _unit);
 _unitType = _unit getVariable ["f_var_assignGear", ""];
+_insigniaVar = _unit getVariable ["f_var_insignia", ""];
 
 _insigniaClass = "";
 
+// Attempt to set insignia from f_var_insignia
+if (!(_insigniaVar isEqualTo "") && (_insigniaClass isEqualTo "" || _insigniaClass isEqualTo [])) then
+{
+    _insigniaClass = DICT_GET(f_dict_insignia_custom,(_insigniaVar));
+};
+
+#ifdef ENABLE_ADVANCED_INSIGNIA
 
-if !(_unitType isEqualTo "") then
+// Attempt to set insignia from unit gearscript role
+if (!(_unitType isEqualTo "" ) && (_insigniaClass isEqualTo "" || _insigniaClass isEqualTo [])) then
```

Comment:
I use `and`, `or` etc instead of `&%`, `||` in this project as a style rule.  Also, consider using `if (A or {B or {C}})` to enable short-circuiting.  SQF doesn't short-circuit by default.